### PR TITLE
[MDS-5798] Request Report

### PR DESCRIPTION
**Pull request by taraepp**
## Objective 
- add a "request report" from for CORE
- fix a couple of bugs I ran into along the way
   - there was a non-action being dispatched which gave me redux errors, changed the name to make it more obvious it was a normal function
   - took comments out of mine report submission response, not necessary
   - submissions did really want an "is_latest" flag, added column and made migrations, adjusted some queries
      - made it really easy to just return the latest submission and not have to do complicated indexing work, so adjusted on FE
      - did that instead of updating the mine report with the submission
- flagged code that should be removed with feature flag

[MDS-5798](https://bcmines.atlassian.net/browse/MDS-5798)

_Why are you making this change? Provide a short explanation and/or screenshots_

### CODE REVIEW

File: services/common/src/components/common/ScrollSidePageWrapper.tsx [hunk 1]
```diff
@@ -18,14 +18,18 @@ const ScrollSidePageWrapper: FC<ScrollSidePageWrapperProps> = ({
...
   const [fixedTop, setIsFixedTop] = useState(false);
   const handleScroll = () => {
-    const scrollHeight = window.scrollY ?? window.pageYOffset;
-    if (scrollHeight > 0 && !fixedTop) {
-      setIsFixedTop(true);
-      // oddly, fixedTop never ends up being true here, even though it seems to take effect,
-      // so took out fixedTop is true out from condition
-    } else if (scrollHeight <= 0) {
-      setIsFixedTop(false);
+    let isMounted = true;
```
Comment (taraepp) [inline services/common/src/components/common/ScrollSidePageWrapper.tsx:21]: console errors, begone!

File: services/common/src/components/forms/RenderCancelButton.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  );
+
+  test("renders properly, does not show confirm on pristine form", () => {
+    const { getByText, container } = render(
+      <ReduxWrapper initialState={changedState}>{renderedForm}</ReduxWrapper>
+    );
+    const button = getByText(buttonText);
+    fireEvent.click(button);
+    expect(cancelFunction).toHaveBeenCalledTimes(0);
+    expect(container).toMatchSnapshot();
```
Comment (taraepp) [inline services/common/src/components/forms/RenderCancelButton.spec.tsx:47]: Something cool I learned here but did not use- it actually generated 2 snapshots in the same file when I kept this line in the other test.

File: services/common/src/components/reports/ReportGetStarted.tsx [hunk 3]
```diff
@@ -35,7 +37,49 @@ interface ReportGetStartedProps {
...
-export const RenderPRRFields: FC<{ mineGuid: string }> = ({ mineGuid }) => {
+export const ReportInfoBox: FC<{ mineReportDefinition: IMineReportDefinition; verb: string }> = ({
+  mineReportDefinition,
+  verb,
+}) => {
+  return (
+    <div className="report-info-box">
+      {mineReportDefinition && (
+        <div>
+          <Typography.Title level={4} className="primary-colour">
```
Comment (taraepp) [inline services/common/src/components/reports/ReportGetStarted.tsx:48]: there was much extracting of functions, but adding primary-colour className was the only real change from the previous version,

File: migrations/sql/V2023.03.11.12.07__add_is_latest_column_to_mine_report_submission.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
+ALTER TABLE mine_report_submission 
+	ADD COLUMN IF NOT EXISTS is_latest boolean DEFAULT TRUE;
+
+DO $$DECLARE report record;
```
Comment (henryoforeh-dev) [inline migrations/sql/V2023.03.11.12.07__add_is_latest_column_to_mine_report_submission.sql:4]: Is "record" used anywhere in the loop?

**Review by henryoforeh-dev — APPROVED** [review]
Looks good, just a small question I had.

File: migrations/sql/V2023.03.11.12.07__add_is_latest_column_to_mine_report_submission.sql (hunk 4 above)
Comment (taraepp) [reply migrations/sql/V2023.03.11.12.07__add_is_latest_column_to_mine_report_submission.sql:4]: assigned in the FOR, referenced in the WHERE. Gotta declare the variable before using it.

File: services/common/src/components/common/ScrollSidePageWrapper.tsx (hunk 1 above)
Comment (matbusby-fw) [reply services/common/src/components/common/ScrollSidePageWrapper.tsx:21]: Take that console errors.  I wish this wasn't necessary though.

File: services/common/src/components/forms/RenderCancelButton.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+interface RenderCancelButtonProps {
+  buttonText?: string | ReactNode;
+  buttonProps?: BaseButtonProps;
+  cancelFunction?: () => void | Promise<void>;
+  cancelModalProps?: ModalFuncProps;
+}
+
+/**
+ * a generic cancel button
```
Comment (matbusby-fw) [inline services/common/src/components/forms/RenderCancelButton.tsx:35]: Woot code comments!

File: services/core-web/src/components/Forms/reports/RequestReportForm.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            />
+          </Col>
+          {selectedReportDefinition && (
+            <Col span={24}>
+              <ReportInfoBox mineReportDefinition={selectedReportDefinition} verb="requesting" />
+            </Col>
+          )}
+        </Row>
+        <Row justify="end">
+          <RenderCancelButton />
```
Comment (matbusby-fw) [inline services/core-web/src/components/Forms/reports/RequestReportForm.tsx:100]: Nice cancel button 👀

**Review by matbusby-fw — APPROVED** [review]
🐊 🥐 🌵 🍰 !

**Pull request merged**